### PR TITLE
Remove remaining FEATURE_CORECLR defines

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Strips out the SecuritySafeCritical and SecurityCritical attributes and the rest of the FEATURE_CORECLR defines.

The first commit is the removal for the attributes and the empty FEATURE_CORECLR defines. (Those that could be matched via simple regex.)

The second commit is the rest of the FEATURE_CORECLR blocks.

I'm in the process of cleaning up the IO code from here. Notably removing the dead security code (which is still allocating and doing unneeded checks).

### CODE REVIEW

File: src/mscorlib/Common/PinnableBufferCache.cs [hunk 1]
```diff
@@ -46,7 +46,7 @@ internal sealed class PinnableBufferCache
 #pragma warning disable 618
         [EnvironmentPermission(SecurityAction.Assert, Unrestricted = true)]
 #pragma warning restore 618
-        [System.Security.SecuritySafeCritical]
```
Comment (jkotas) [inline src/mscorlib/Common/PinnableBufferCache.cs]: Could you please fix your regex to not leave empty line behind after these?

File: src/mscorlib/Common/PinnableBufferCache.cs [hunk 2]
```diff
@@ -46,7 +46,7 @@ internal sealed class PinnableBufferCache
 #pragma warning disable 618
         [EnvironmentPermission(SecurityAction.Assert, Unrestricted = true)]
```
Comment (jkotas) [inline src/mscorlib/Common/PinnableBufferCache.cs]: BTW: These Permission attributes can be stripped too - they do not do anything in CoreCLR.

**Comment by jkotas** [comment]
Some SecurityCriticialAttributes are in the contracts in corefx (e.g. look at System.Runtime contract). Could you please check that the contract validation won't complain?

We may want to remove them from the contracts too.

**Comment by stephentoub** [comment]
> We may want to remove them from the contracts too.

These attributes are never verified by the language compiler, right?  If so, I agree it makes sense to remove them from the contracts.  I actually thought we already had, but maybe we only did from some of them.

File: src/mscorlib/Common/PinnableBufferCache.cs (hunk 2 above)
Comment (JeremyKuhne) [inline src/mscorlib/Common/PinnableBufferCache.cs]: I'll pull them in another PR

File: src/mscorlib/Common/PinnableBufferCache.cs (hunk 1 above)
Comment (JeremyKuhne) [inline src/mscorlib/Common/PinnableBufferCache.cs]: Actually only this file has the line feeds, that's odd. I'll fix this one.

Comment (JeremyKuhne) [inline src/mscorlib/Common/PinnableBufferCache.cs]: Argh- I see what I did. Copied one of the regexes incorrectly. I'll update and pull the other security attributes as well. I'll put the regex up for everyone to look at.

**Comment by JeremyKuhne** [comment]
Ok, I've merged in #8571 to this change.

**Comment by JeremyKuhne** [comment]
I've yanked the attributes in CoreFX as well:

https://github.com/dotnet/corefx/pull/14383

The tools didn't seem to care about the attribute matching.

**Comment by JeremyKuhne** [comment]
@dotnet-bot test this please

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Linux ARM Emulator Cross Debug Build

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Linux ARM Emulator Cross Debug Build (CI is having issues)

**Pull request merged**